modules/@apostrophecms/piece-type/ui/apos/components/AposRelationshipEditor.vue

```diff
@@ -79,13 +80,16 @@ export default {
   async mounted() {
     this.modal.active = true;
     this.docReady = true;
-    this.doc.data = this.value || {};
   },
   methods: {
     async submit() {
-      this.$emit('input', this.doc.data);
-      this.cancel();
+      this.$emit('input', this.docInfo.data);
+      this.modal.showModal = false;
```

Comment:
This should go back to `cancel` when the other logic is moved into the `AposEditorMixin` file.